src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -9475,54 +9533,59 @@ namespace SurfaceGeometry {
-                    if (itnext == std::end(vertices)) {
-                        itnext = std::begin(vertices);
-                    }
+                auto it = vertices.begin();
+                std::advance(it, popResult.poppedVertexPos);
+                int const poppedVertexIndex = popResult.poppedVertexPos + 1;
 
-                    // TODO: use isAlmostEqual3Pt for consistency? (which uses 0.0127 m / 1/2inch instead of 0.01 m)
-                    DistanceCheck = distance(*it, *itnext);
-                    if (DistanceCheck < 0.01) {
-                        int curVertexIndex = std::distance(vertices.begin(), it) + 1;
-                        int nextVertexIndex = std::distance(vertices.begin(), itnext) + 1;
-                        if (state.dataGlobal->DisplayExtraWarnings) {
-                            ShowWarningError(state,
-                                             format("{}Distance between two vertices < .01, possibly coincident. for Surface={}, in Zone={}",
-                                                    RoutineName,
-                                                    state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Name,
-                                                    state.dataSurfaceGeometry->SurfaceTmp(SurfNum).ZoneName));
-                            ShowContinueError(state, format("Vertex [{}]=({:.2R},{:.2R},{:.2R})", curVertexIndex, it->x, it->y, it->z));
-                            ShowContinueError(state, format("Vertex [{}]=({:.2R},{:.2R},{:.2R})", nextVertexIndex, itnext->x, itnext->y, it->z));
-                        }
-                        ++state.dataErrTracking->TotalCoincidentVertices;
-                        if (nSides > 3) {
-                            if (state.dataGlobal->DisplayExtraWarnings) {
-                                ShowContinueError(state, format("Dropping Vertex [{}].", nextVertexIndex));
-                            }
-                            --nSides;
-                            vertices.erase(itnext);
-                            poppedVertex = true;
-                            break;
-                        } else {
-                            if (state.dataGlobal->DisplayExtraWarnings) {
-                                ShowContinueError(state,
-                                                  format("Cannot Drop Vertex [{}]; Number of Surface Sides at minimum. This surface is now a "
-                                                         "degenerate surface.",
-                                                         curVertexIndex));
-                            }
-                            ++state.dataErrTracking->TotalDegenerateSurfaces;
-                            // mark degenerate surface?
-                        }
+                auto itKept = vertices.begin();
+                std::advance(itKept, popResult.keptVertexPos);
+                int const keptVertexIndex = popResult.keptVertexPos + 1;
```

Comment:
Grab the popped one, and the kept one (regardless of whether it's previous or next)